api/libproject/devstart.rb

```diff
@@ -375,65 +389,35 @@ def get_auth_login_account()
   return `gcloud config get-value account`.strip()
 end
 
-# Fetches a credentials file. Passes the path of the credentials to a block.
-# For all-of-us-workbench-test only, it leaves the (lazy-fetched) creds on disk;
-# for any other project, it cleans them up after the block is run.
-def do_run_with_creds(project, account, creds_file)
-  if creds_file == nil
-    service_account_creds_file = Tempfile.new("#{project}-creds.json")
-    if project == "all-of-us-workbench-test"
-      creds_filename = "src/main/webapp/WEB-INF/sa-key.json"
-      # For test, use a locally stored key file copied from GCS (which we leave hanging
-      # around.)
-      if !File.file?(creds_filename)
-        # Create a temporary creds file for accessing GCS.
-        get_service_account_creds_file(project, account, service_account_creds_file)
-        begin
-          activate_service_account(service_account_creds_file.path)
-          # Copy the stable creds file from its path in GCS to sa-key.json.
-          # Important: we must leave this key file in GCS, and not delete it in Cloud Console,
-          # or local development will stop working.
-          get_file_from_gcs("all-of-us-workbench-test-credentials",
-              "all-of-us-workbench-test-9b5c623a838e.json", creds_filename)
-        ensure
-          # Delete the temporary creds we created.
-          delete_service_accounts_creds(project, account, service_account_creds_file)
-        end
-      end
-      yield(creds_filename)
-    else
-      # Create a creds file and use it; clean up when done.
-      get_service_account_creds_file(project, account, service_account_creds_file)
-      begin
-        yield(service_account_creds_file.path)
-      ensure
-        delete_service_accounts_creds(project, account, service_account_creds_file)
-      end
-    end
-  else
-    yield(creds_file)
-  end
-end
-
-def register_service_account(*args)
-  GcloudContext.new("register-service-account", args).run do |ctx|
+def register_service_account(cmd_name, *args)
+  ensure_docker cmd_name, args
+  op = WbOptionsParser.new(cmd_name, args)
+  gcc = GcloudContextV2.new(op)
+  op.parse.validate
+  gcc.validate
+  ServiceAccountContext.new(gcc.project).run do
```

Comment:
I don't understand what it means to "run" a ServiceAccountContext ("running" a SqlCloudProxy does make sense to me). I would probably choose `.with` over `.run`.

Reply:
Will address in a follow-up.